### DISCUSSION TITLE
to marriage sumatrapdfreader & dnGrep

### DISCUSSION THREAD

**Discussion by Aliaksandr3by** [sumatrapdfreader/sumatrapdf#3555] [Ideas]
get line number from dnGrep and sent it  to sumatrapdf .

the pattern %line give to you a line number, but -page  get page number

![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/11432724/6c262884-3cc8-48b5-9e27-3f65b8a44683)

**Comment by GitHubRulesOK** [comment]
I dont see how that can work for a PDF since there are no line numbers in a PDF the Top line on a page can be 6 then 5 and the second line down be 100 +4+99 since PDF structure is randomly built up from single letters. 
How it is done in LaTeX is via scrolled area so in effect PDFsync or SynTeX is goto -scroll x.y

For flowable like DocX or ePub it is slightly different the lines are sequential so top 2 lines may be 1+2+3+4 or simply half of line one.

Few formats actually have fixated line numbers (nor real page numbers) so DnGrep decides what those numbers may be in its repertoire not actually are in source.

Sumatra does have search forward so you can search page 2 IF the word is a regular string

**Reply by Aliaksandr3by** [reply]
ok I understand.
bad, why dnGrep don't have a page number veriable.
https://github.com/dnGrep/dnGrep/wiki/Custom-Editor-Integration

**Reply by GitHubRulesOK** [reply]
Most editors dont use pages until OUTPUT so input is by line number and out by page number 
so line 70 could be page 1 or 2 or any other
(SumatraPDF is a page output reader so uses pages for input)
